packages/account/src/predicate/predicate.ts

```diff
@@ -40,27 +48,32 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
   /**
    * Creates an instance of the Predicate class.
    *
-   * @param bytes - The bytes of the predicate.
+   * @param bytecode - The bytecode of the predicate.
+   * @param abi - The JSON ABI of the predicate.
    * @param provider - The provider used to interact with the blockchain.
-   * @param jsonAbi - The JSON ABI of the predicate.
+   * @param inputData - The predicate input data (optional).
    * @param configurableConstants - Optional configurable constants for the predicate.
    */
-  constructor(
-    bytes: BytesLike,
-    provider: Provider,
-    jsonAbi?: JsonAbi,
-    configurableConstants?: { [name: string]: unknown }
-  ) {
+  constructor({
+    bytecode,
+    abi,
+    provider,
+    inputData,
+    configurableConstants,
+  }: PredicateConstructorArgs<ARGS>) {
```

Comment:
Not your doing, but here's another suggestion if you like it:

```suggestion
  }: PredicateConstructorArgs<TInputData>) {
```